src/Compilers/Core/Portable/Serialization/ObjectWriter.cs

```diff
@@ -483,7 +483,8 @@ private void WriteArray(Array array)
                     // don't blow the stack.  'LongRunning' ensures that we get a dedicated thread
                     // to do this work.  That way we don't end up blocking the threadpool.
                     var task = Task.Factory.StartNew(
-                        () => WriteArrayValues(array),
+                        a => WriteArrayValues((Array)a), 
```

Comment:
@CyrusNajmabadi and I spoke offline. There's still an allocation of a delegate - it doesn't appear to be showing at all on the radar. I looked at the code gen and figured out why:

- The first thing that the compiler does in the method is instantiate the capture class and "capture" _array_.
- It then replaces all reads/writes of the parameter to the field in the capture class

Basically, in summary it looks like this method is always paying for the capture class regardless of the branch. That's why the delegate creation itself isn't showing.